apache/trafficcontrol
# TP: adds the ability to rearrange delivery service table columns as well as toggle column visibility/searchability

## What does this PR (Pull Request) do?


This PR adds checkboxes to toggle the visiblity/searchability of ds table columns as well as adds the ability to reorder the columns.

![image](https://user-images.githubusercontent.com/251272/61484811-e5437700-a95c-11e9-907f-b93cbbb1cff9.png)

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?
1. Run the [UI tests](https://github.com/apache/trafficcontrol/tree/master/traffic_portal/test/end_to_end)
2. Navigate to https://tp.doman.com/#!/delivery-services and

1. reorder the columns; refresh the screen to ensure that the new order persists itself across browser sessions.
2. click on the drop down to toggle the visiblity/searchability of columns; refresh the screen to ensure that the selected columns are visible/hidden across browser sessions. also, ensure that values in hidden columns are not considered in the table search.

## The following criteria are ALL met by this PR
